### PR TITLE
Ssh OIDC marcus

### PR DESCRIPTION
**Pull request by marcvs**
# Summary

Add documentation about ssh-oidc

### CODE REVIEW

File: content/en/users/compute/cloud-compute/auth/_index.md [hunk 1]
```diff
@@ -151,5 +151,58 @@ export OS_TOKEN=$(fedcloud openstack --site <NAME_OF_SITE> --vo <NAME_OF_VO> \
 
 ### Using ssh-oidc
 
-[ssh-oidc](https://github.com/EOSC-synergy/ssh-oidc) is a set of tools that
-allows ssh with OIDC.
+[ssh-oidc](https://github.com/EOSC-synergy/ssh-oidc) is a collection of
+tools that provide enable ssh, using OpenID-Connect access tokens.
```
Comment (sebastian-luna-valero) [inline content/en/users/compute/cloud-compute/auth/_index.md:155]:
```suggestion
tools that enable ssh using OpenID-Connect access tokens.
```

File: content/en/users/compute/cloud-compute/auth/_index.md [hunk 2]
```diff
@@ -151,5 +151,58 @@ export OS_TOKEN=$(fedcloud openstack --site <NAME_OF_SITE> --vo <NAME_OF_VO> \
...
+- Do not store state whenever possible
+  - Single exception: federated-user -> local-user mapping is stored in `passwd`
+- Small components that work individually (one tool for one job)
+
+#### Installation (client-side)
+
+`ssh-oidc` does not require specific clients per se. OIDC Access-Tokens
+may simply be passed, when ssh prompts for the "Access Token".
+Unfortunately, the most-popular ssh-client does not support access-tokens
+which are longer than 1024 byte _and_ EGI uses tokens that are typically
```
Comment (sebastian-luna-valero) [inline content/en/users/compute/cloud-compute/auth/_index.md:173]:
```suggestion
which are longer than 1024 bytes _and_ EGI uses tokens that are typically
```

File: content/en/users/compute/cloud-compute/auth/_index.md [hunk 3]
```diff
@@ -151,5 +151,58 @@ export OS_TOKEN=$(fedcloud openstack --site <NAME_OF_SITE> --vo <NAME_OF_VO> \
...
+longer than 1024 byte.
```
Comment (sebastian-luna-valero) [inline content/en/users/compute/cloud-compute/auth/_index.md:174]:
```suggestion
longer than 1024 bytes.
```

File: content/en/users/compute/cloud-compute/auth/_index.md [hunk 4]
```diff
@@ -151,5 +151,58 @@ export OS_TOKEN=$(fedcloud openstack --site <NAME_OF_SITE> --vo <NAME_OF_VO> \
...
+
+Hence, we suggest using [mccli](https://mccli.readthedocs.org) (`pip
+install mccli`) on Unix/Mac. Windows users should use the putty
+`oidc-plugin`, which conveniently ships with
+[oidc-agent-for-windows](http://repo.data.kit.edu/windows/oidc-agent/)
```
Comment (sebastian-luna-valero) [inline content/en/users/compute/cloud-compute/auth/_index.md:179]:
```suggestion
[oidc-agent-for-windows](http://repo.data.kit.edu/windows/oidc-agent/).
```

File: content/en/users/compute/cloud-compute/auth/_index.md [hunk 5]
```diff
@@ -151,5 +151,58 @@ export OS_TOKEN=$(fedcloud openstack --site <NAME_OF_SITE> --vo <NAME_OF_VO> \
...
+
+For conveniently obtaining access tokens,
```
Comment (sebastian-luna-valero) [inline content/en/users/compute/cloud-compute/auth/_index.md:181]:
```suggestion
For obtaining access tokens,
```

File: content/en/users/compute/cloud-compute/auth/_index.md [hunk 6]
```diff
@@ -151,5 +151,58 @@ export OS_TOKEN=$(fedcloud openstack --site <NAME_OF_SITE> --vo <NAME_OF_VO> \
...
+[oidc-agent](https://indigo-dc.gitbook.io/oidc-agent) is probably the most
+convenient choice.
+
+#### Installation (server-side)
+
+Within the federated cloud, there are multiple options for installing
+the ssh-oidc server-side components
```
Comment (sebastian-luna-valero) [inline content/en/users/compute/cloud-compute/auth/_index.md:188]:
```suggestion
the ssh-oidc server-side components:
```

File: content/en/users/compute/cloud-compute/auth/_index.md [hunk 7]
```diff
@@ -151,5 +151,58 @@ export OS_TOKEN=$(fedcloud openstack --site <NAME_OF_SITE> --vo <NAME_OF_VO> \
...
+
+1. Via the Infrastructure Manager (IM) <https://im.egi.eu>: IM offers
+   starting a variety of VM infrastructures.
```
Comment (sebastian-luna-valero) [inline content/en/users/compute/cloud-compute/auth/_index.md:191]:
```suggestion
1. Via the [Infrastructure Manager (IM)](https://im.egi.eu):
```

File: content/en/users/compute/cloud-compute/auth/_index.md [hunk 8]
```diff
@@ -151,5 +151,58 @@ export OS_TOKEN=$(fedcloud openstack --site <NAME_OF_SITE> --vo <NAME_OF_VO> \
...
+   - Choosing the "virtual machine with extra HD".
```
Comment (sebastian-luna-valero) [inline content/en/users/compute/cloud-compute/auth/_index.md:192]:
```suggestion
   - Choosing the `virtual machine with extra HD`.
```

File: content/en/users/compute/cloud-compute/auth/_index.md [hunk 9]
```diff
@@ -151,5 +151,58 @@ export OS_TOKEN=$(fedcloud openstack --site <NAME_OF_SITE> --vo <NAME_OF_VO> \
...
+
+   The started VM will then allow ssh access with an OIDC Access Token.
+
+   The user that started the VM will be mapped to `cloudadm` and be able to execute
+   `sudo`.
+
+1. Manual installation:
+    - Install `motley-cue`, and `pam-ssh-oidc` (or
+      `pam-ssh-oidc-autoconfig`). Details here:
+          <https://github.com/EOSC-synergy/ssh-oidc/blob/master/installation.md>
```
Comment (sebastian-luna-valero) [inline content/en/users/compute/cloud-compute/auth/_index.md:205]:
```suggestion
      `pam-ssh-oidc-autoconfig`). Details in the [ssh-oid repository](https://github.com/EOSC-synergy/ssh-oidc/blob/master/installation.md).
```

File: content/en/users/compute/cloud-compute/auth/_index.md [hunk 10]
```diff
@@ -151,5 +151,58 @@ export OS_TOKEN=$(fedcloud openstack --site <NAME_OF_SITE> --vo <NAME_OF_VO> \
...
+    - Configure `motley-cue`. Details here:
+        <https://github.com/EOSC-synergy/ssh-oidc/blob/master/configuration.md>
+
```
Comment (sebastian-luna-valero) [inline content/en/users/compute/cloud-compute/auth/_index.md:208]:
```suggestion
    - Configure `motley-cue`. See example in the [ssh-oidc repository]https://github.com/EOSC-synergy/ssh-oidc/blob/master/configuration.md).
```

**Review by sebastian-luna-valero — APPROVED** [review]
Many thanks!

I have proposed a few changes. Feel free to add/reject them!

File: content/en/users/compute/cloud-compute/auth/_index.md [hunk 11]
```diff
@@ -151,5 +151,55 @@ export OS_TOKEN=$(fedcloud openstack --site <NAME_OF_SITE> --vo <NAME_OF_VO> \
...
+   The started VM will then allow ssh access with an OIDC Access Token.
+
+   The user that started the VM will be mapped to `cloudadm` and be able to execute
+   `sudo`.
+
+1. Manual installation:
+    - Install `motley-cue`, and `pam-ssh-oidc` (or
+      `pam-ssh-oidc-autoconfig`). Details in the [ssh-oid repository](https://github.com/EOSC-synergy/ssh-oidc/blob/master/installation.md).
+    - Configure `motley-cue`. See example in the [ssh-oidc repository]https://github.com/EOSC-synergy/ssh-oidc/blob/master/configuration.md).
```
Comment (gwarf) [inline content/en/users/compute/cloud-compute/auth/_index.md:205]:
```suggestion
    - Configure `motley-cue`. See example in the [ssh-oidc repository](https://github.com/EOSC-synergy/ssh-oidc/blob/master/configuration.md).
```

File: content/en/users/compute/cloud-compute/auth/_index.md [hunk 12]
```diff
@@ -151,5 +151,55 @@ export OS_TOKEN=$(fedcloud openstack --site <NAME_OF_SITE> --vo <NAME_OF_VO> \
...
+Most of the individual tools work standalone, but they may be combined to
+establish complex installations with dynamic account provisioning in
+multi-OP and multi-VO contexts.
+
+The tools share these common design criteria:
+
+- None of the ssh-client or ssh-server components will be modified
+- Do not store state whenever possible
+    - Single exception: federated-user -> local-user mapping is stored in
+        `passwd`
```
Comment (gwarf) [inline content/en/users/compute/cloud-compute/auth/_index.md]:
```suggestion
- Do not store state whenever possible
  - Single exception: federated-user -> local-user mapping is stored in
    `passwd`
```

**Comment by gwarf** [comment]
Thanks!
Let's bypass the errors on www.egi.eu for now, some shady things are currently happening with the EGI site.

**Comment by gwarf** [comment]
@enolfc, does it look OK to you?

File: content/en/users/compute/cloud-compute/auth/_index.md [hunk 13]
```diff
@@ -151,5 +151,55 @@ export OS_TOKEN=$(fedcloud openstack --site <NAME_OF_SITE> --vo <NAME_OF_VO> \
...
 ### Using ssh-oidc
 
-[ssh-oidc](https://github.com/EOSC-synergy/ssh-oidc) is a set of tools that
-allows ssh with OIDC.
+[ssh-oidc](https://github.com/EOSC-synergy/ssh-oidc) is a collection of
+tools that enable ssh using OpenID-Connect access tokens.
+
+Most of the individual tools work standalone, but they may be combined to
+establish complex installations with dynamic account provisioning in
+multi-OP and multi-VO contexts.
```
Comment (enolfc) [inline content/en/users/compute/cloud-compute/auth/_index.md:159]: what is multi-OP?

File: content/en/users/compute/cloud-compute/auth/_index.md [hunk 14]
```diff
@@ -151,5 +151,55 @@ export OS_TOKEN=$(fedcloud openstack --site <NAME_OF_SITE> --vo <NAME_OF_VO> \
...
+longer than 1024 bytes.
+
+Hence, we suggest using [mccli](https://mccli.readthedocs.org) (`pip
+install mccli`) on Unix/Mac. Windows users should use the putty
+`oidc-plugin`, which conveniently ships with
+[oidc-agent-for-windows](http://repo.data.kit.edu/windows/oidc-agent/).
+
+For conveniently obtaining access tokens,
+[oidc-agent](https://indigo-dc.gitbook.io/oidc-agent) is probably the most
+convenient choice.
```
Comment (enolfc) [inline content/en/users/compute/cloud-compute/auth/_index.md:184]: conveniently and convenient in the same sentence?

**Review by enolfc — COMMENTED** [review]
Thanks @marcvs. Just 2 minor comments

File: content/en/users/compute/cloud-compute/auth/_index.md (hunk 13 above)
Comment (marcvs) [reply content/en/users/compute/cloud-compute/auth/_index.md:159]: Supporting more than a single OP (OIDC-Provider). I.e. it's possible to support EGI-Checkin and Indigo-IAM at the same time.
Shall I extend that description?

File: content/en/users/compute/cloud-compute/auth/_index.md (hunk 14 above)
Comment (marcvs) [reply content/en/users/compute/cloud-compute/auth/_index.md:184]: Well, oidc-agent is super convenient :)
I'll change that with the above comment.

File: content/en/users/compute/cloud-compute/auth/_index.md (hunk 13 above)
Comment (gwarf) [reply content/en/users/compute/cloud-compute/auth/_index.md:159]: yes, please, not sure we are using this term anywhere in the doc, we are Check-in powered you know ;)

**Review by enolfc — APPROVED** [review]
LGTM!

**Pull request merged**